src/liballoc/raw_vec.rs

```diff
@@ -52,15 +52,18 @@ impl<T, A: Alloc> RawVec<T, A> {
     /// Like `new`, but parameterized over the choice of allocator for
     /// the returned `RawVec`.
     pub const fn new_in(a: A) -> Self {
-        // `!0` is `usize::MAX`. This branch should be stripped at compile time.
-        // FIXME(mark-i-m): use this line when `if`s are allowed in `const`:
-        //let cap = if mem::size_of::<T>() == 0 { !0 } else { 0 };
+        let cap = {
+            #[cfg(not(bootstrap))]
+            { if mem::size_of::<T>() == 0 { !0 } else { 0 } }
+
+            #[cfg(bootstrap)]
+            [0, !0][(mem::size_of::<T>() == 0) as usize]
+        };
```

Comment:
Don't we still want the `!0 is usize::MAX` part of the comment here (or we could just use `usize::MAX`)? Also we should leave a `FIXME` reminder to take out the `cfg(bootstrap)` stuff. As an aside, do we want to wait until the bootstrap compiler gets `#![feature(const_if_match)]` to make this change?

Reply:
Honestly, we could do that. We will have to wait a cycle to remove the `bootstrap` stuff anyway...